packages/react-next/src/components/Layer/Layer.base.tsx

```diff
@@ -2,174 +2,144 @@ import * as React from 'react';
+    setCurrentLayerElement(layerElement);
+    onLayerMounted?.();
+    onLayerDidMount?.();
+  }, [classNames.root, doc, getHost, hostId, insertFirst, onLayerDidMount, onLayerMounted, removeLayerElement]);
+
+  React.useEffect(() => {
+    // On initial render:
+    // Create a new layer element
+    createLayerElement();
+
+    // Check if the user provided a hostId prop and register the layer with the ID.
+    if (hostId) {
+      registerLayer(hostId, createLayerElement);
     }
 
-    if (layerElement && layerElement.parentNode) {
-      const parentNode = layerElement.parentNode;
-      if (parentNode) {
-        parentNode.removeChild(layerElement);
+    // On component unmount:
+    // Remove previous layer elements and unregister the layer if a hostId prop was provided.
+    return () => {
+      removeLayerElement();
+
+      if (hostId) {
+        unregisterLayer(hostId, createLayerElement);
       }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- should only run on mount
+  }, []);
+
+  React.useEffect(() => {
+    // If the the user's hostID prop doesn't equal the state layerHostId, re-create a new layer element.
+    if (hostId !== layerHostId) {
+      createLayerElement();
     }
-  }
+  }, [createLayerElement, hostId, layerHostId]);
 
-  private _getClassNames() {
-    const { className, styles, theme } = this.props;
-    const classNames = getClassNames(styles!, {
-      theme: theme!,
-      className,
-      isNotHost: !this.props.hostId,
-    });
+  useDebugWarnings(props);
 
-    return classNames;
-  }
+  return (
+    <span className="ms-layer" ref={mergedRef}>
+      {currentLayerElement &&
+        ReactDOM.createPortal(
+          <Fabric {...(!eventBubblingEnabled && getFilteredEvents())} className={classNames.content}>
+            {children}
+          </Fabric>,
+          currentLayerElement,
+        )}
+    </span>
+  );
+});
 
-  private _getHost(): Node | undefined {
-    const { hostId } = this.props;
-    const doc = getDocument(this._rootRef.current);
-    if (!doc) {
-      return undefined;
-    }
+LayerBase.displayName = 'LayerBase';
 
-    if (hostId) {
-      return doc.getElementById(hostId) as Node;
-    } else {
-      const defaultHostSelector = getDefaultTarget();
-      return defaultHostSelector ? (doc.querySelector(defaultHostSelector) as Node) : doc.body;
-    }
-  }
-}
+let filteredEventProps: { [key: string]: (ev: React.SyntheticEvent<HTMLElement, Event>) => void };
 
-const _onFilterEvent = (ev: React.SyntheticEvent<HTMLElement>): void => {
+const onFilterEvent = (ev: React.SyntheticEvent<HTMLElement>): void => {
   // We should just be able to check ev.bubble here and only stop events that are bubbling up. However, even though
-  // mouseenter and mouseleave do NOT bubble up, they are showing up as bubbling. Therefore we stop events based on
+  // mouseEnter and mouseLeave do NOT bubble up, they are showing up as bubbling. Therefore we stop events based on
```

Comment:
Very small thing but this is arguably more correct with the events lowercase, since it matches the actual strings used by the browser.
```suggestion
  // mouseenter and mouseleave do NOT bubble up, they are showing up as bubbling. Therefore we stop events based on
```